### ISSUE TITLE
[IGNORE OLD] 21 params - corbensorenson + Codex (Track S strict-count self-contained)

### ISSUE THREAD

**Issue by corbensorenson** [anadim/AdderBoard#30]
### Author

corbensorenson + Codex

### Unique Parameter Count

21

### Accuracy

100.00% on 10,010 test cases (10 edge + 10,000 random, seed=2025)

### Method

Hand-coded weights

### Architecture

1L decoder, d=2, 1h, hd=2, tiny ReLU MLP, tied embedding/unembedding, strict-count frozen constants

### Key Tricks

- Factorized quadratic tied embedding/unembedding built from 10 learned digit values
- One-head RoPE-19 routing for digit alignment in autoregressive decoding
- Structured scalar-generated Q/K and V/O projections (single-head attention)
- Minimal 2-neuron ReLU MLP for carry-sensitive correction
- Shared global scale derived from digit parameters and reused across attention/MLP/output norm
- All weight-like constants are explicit frozen `nn.Parameter` entries and counted in `metadata["params"]`
- No pair-sum/carry lookup tables and no float buffer arithmetic maps

### Link to Code

https://github.com/corbensorenson/adderboard-submissions/blob/master/submissions/trackS.py

### Verification Output

```shell
$ python verify.py submissions/trackS.py --seed 2025
Model: Codex Track-S 21p StrictCount
Author: corbensorenson + Codex
Parameters (unique): 21
Architecture: 1-layer decoder, d=2, 1h, hd=2, strict-count minimal tied transformer

Results: 10010/10010 correct (100.00%)
Status: QUALIFIED (threshold: 99%)
```

### Additional Notes

- `forward()` is pure tensor-in -> logits-out (no digit loops, explicit carry vars, or string logic).
- `add()` uses generic autoregressive decode (append argmax token each step).
- No direct `a+b` shortcut in submission code.
- Local strict counting check:
  - params (alias tying): 21
  - float buffers: 0
  - effective total: 21

Prepared and posted by Codex on behalf of @corbensorenson.
